File: class 3/2606.py
import sys

x = int(sys.stdin.readline())
y = int(sys.stdin.readline())

dic = {}

def dfs(start, dic) :
    for i in dic[start] :
        if i not in visited :
            visited.append(i)
            dfs(i, dic)

for i in range(x):
    dic[i+1] = set()

for j in range(y) :
    a, b = map(int, sys.stdin.readline().split())
    dic[a].add(b)
    dic[b].add(a)

visited = []
dfs(1, dic)

print(len(visited)-1)

File: 백준 문제 풀이 모음/kotlin/class 3/2606.kt
package class3

var connect = Array(0) { MutableList(0) { 0 } }
var visited = Array(0) { false }

val dfs = mutableListOf<Int>()
fun main() = with(System.`in`.bufferedReader()) {
    val computer = readLine().toInt()
    val pair = readLine().toInt()

    connect = Array(computer+1) { MutableList(0) { 0 } }
    visited = Array(computer+1) { false }
    repeat(pair) {
        val input = readLine().split(' ').toList()
        val a = input[0].toInt()
        val b = input[1].toInt()

        connect[a].add(b)
        connect[b].add(a)

    }

    DFS(1)

    print(dfs.size-1)
//    connect.forEach {
//        println(it)
//    }



}

fun DFS(i: Int) {
    dfs.add(i)
    visited[i] = true

    connect[i].forEach {
        if(!visited[it]) {
            DFS(it)
        }
    }
}
